app/src/hooks/useTokens.tsx

```diff
@@ -51,25 +53,25 @@ export const useTokens = (
   relay?: boolean,
   addBridgeAllowance?: boolean,
 ) => {
-  const defaultTokens = getTokensByNetwork(context.networkId)
-  if (addNativeAsset) {
-    defaultTokens.unshift(getNativeAsset(context.networkId, relay))
+  let defaultTokens: Token[] = []
+  if (context) {
+    defaultTokens = getTokensByNetwork(context.networkId)
+    if (addNativeAsset) {
+      defaultTokens.unshift(getNativeAsset(context.networkId, relay))
+    }
   }
   const [tokens, setTokens] = useState<Token[]>(defaultTokens)
 
-  const omenTCRListId = getOmenTCRListId(context.networkId)
-
-  const { data, error, loading, refetch } = useQuery<GraphResponse>(query, {
-    notifyOnNetworkStatusChange: true,
-    skip: false,
-    variables: { listId: omenTCRListId.toString() },
-  })
+  const fetchTokenDetails = async () => {
+    if (context) {
+      const omenTCRListId = getOmenTCRListId(context.networkId)
+      const { httpUri } = getGraphUris(context.networkId)
+      const variables = { listId: omenTCRListId.toString() }
+      const response = await axios.post(httpUri, { query, variables })
+      const data: GraphResponse = response.data.data
```

Comment:
What's the reasoning for switching from `useQuery` to `axios`?

Reply:
This was an annoying dependency tree issue, `useQuery` requires [`ApolloProviderWrapper`](https://github.com/protofire/omen-exchange/blob/master/app/src/App.tsx#L47) which requires [`ConnectedWeb3`](https://github.com/protofire/omen-exchange/blob/master/app/src/contexts/Apollo.tsx#L8) and because we are now using `useTokens` inside `ConnectedWeb3` we don't have access to the `ApolloProviderWrapper` yet. I attempted to swap the providers around but just got the same issue in reverse lol

Reply:
Another issue was that hooks can't be used conditionally which was needed here if the connection context had not been defined yet.